### ISSUE TITLE
libx264 height not divisible by 2 (640x359)

### ISSUE THREAD

**Issue by yondonfu** [livepeer/go-livepeer#354]
```
[libx264 @ 0x7f7f74100540] height not divisible by 2 (640x359)
Error opening video encoder
transcoder: Unable to open output
fatal error: unexpected signal during runtime execution
[signal SIGSEGV: segmentation violation code=0x1 addr=0x7fffffff0 pc=0xe68bf7]

runtime stack:
runtime.throw(0x150726f, 0x2a)
        /usr/local/go/src/runtime/panic.go:605 +0x95
runtime.sigpanic()
        /usr/local/go/src/runtime/signal_unix.go:351 +0x2b8

goroutine 28840 [syscall, locked to thread]:
runtime.cgocall(0xda0e30, 0xc4227ef668, 0x1505923)
        /usr/local/go/src/runtime/cgocall.go:132 +0xe4 fp=0xc4227ef628 sp=0xc4227ef5e8 pc=0x443184
github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg._Cfunc_lpms_transcode(0x7f7f740008c0, 0xc4213ad580, 0x2, 0x0)
        github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg/_obj/_cgo_gotypes.go:188 +0x4d fp=0xc4227ef668 sp=0xc4227ef628 pc=0x8505dd
github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg.Transcode.func1(0x7f7f740008c0, 0xc4213ad580, 0x2, 0x0)
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg/ffmpeg.go:64 +0x72 fp=0xc4227ef6a0 sp=0xc4227ef668 pc=0x8517b2
github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg.Transcode(0xc4200aeb40, 0x43, 0x7fff2dcd9595, 0x2b, 0xc422cfa2d0, 0x2, 0x2, 0x0, 0xc4227ef908)
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/ffmpeg/ffmpeg.go:64 +0x5e6 fp=0xc4227ef870 sp=0xc4227ef6a0 pc=0x850f26
github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/transcoder.(*FFMpegSegmentTranscoder).Transcode(0xc4238c5100, 0xc4200aeb40, 0x43, 0xe5ff4, 0xe5ff4, 0x1a4, 0x0, 0x0)
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/vendor/github.com/livepeer/lpms/transcoder/ffmpeg_segment_transcoder.go:27 +0x97 fp=0xc4227ef9a0 sp=0xc4227ef870 pc=0xd749e7
github.com/livepeer/go-livepeer/core.(*LivepeerNode).transcodeAndBroadcastSeg(0xc4201186c0, 0xc4227eff08, 0xc4200ae9b0, 0x41, 0x41, 0x203cea0, 0xc4207b4280, 0x20299c0, 0xc4238c5100, 0xc422b038e0, ...)
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/core/livepeernode.go:274 +0x2cc fp=0xc4227efc30 sp=0xc4227ef9a0 pc=0xd77a5c
github.com/livepeer/go-livepeer/core.(*LivepeerNode).TranscodeAndBroadcast.func1(0x0, 0xc424310000, 0xe6158, 0xe6158, 0xc4210ace00)
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/core/livepeernode.go:244 +0x826 fp=0xc4227effb8 sp=0xc4227efc30 pc=0xd7bd96
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2337 +0x1 fp=0xc4227effc0 sp=0xc4227effb8 pc=0x49bc31
created by github.com/livepeer/go-livepeer/vendor/github.com/livepeer/go-livepeer-basicnet.(*BasicSubscriber).startWorker.func1
        /home/vagrant/go/src/github.com/livepeer/go-livepeer/vendor/github.com/livepeer/go-livepeer-basicnet/basic_subscriber.go:114 +0xf3
```

Observed in both devenv and streaming into ingest server with OBS.

Here are my OBS settings:

<img width="1249" alt="screen shot 2018-04-03 at 6 08 09 pm" src="https://user-images.githubusercontent.com/5933273/38278450-13cf8a0c-376a-11e8-9b61-a3de58f8e507.png">

<img width="1179" alt="screen shot 2018-04-03 at 6 08 32 pm" src="https://user-images.githubusercontent.com/5933273/38278454-16c069fc-376a-11e8-9d63-4f9cd3808094.png">

**Comment by j0sh** [comment]
Pretty sure the segfault is fixed by https://github.com/livepeer/lpms/pull/64 ; in fact, the 'odd height' is the same scenario that's used to test the fix.

The fix has been merged into LPMS, and will be in goclient once https://github.com/livepeer/go-livepeer/pull/351 is merged.

**Comment by yondonfu** [comment]
Segfault seems to be fixed by @josh's branch in local testing. I still got a `Media length check failed` error likely because the segment length checker enforces a segment length of 4 seconds while the default segment length defined in go-livepeer on this line https://github.com/livepeer/go-livepeer/blob/8194501fa3f0514a8b1589edfc19ce2ff33a36cc/server/mediaserver.go#L39 is 8 seconds - changing the segment length to 4 seconds fixed the error

**Comment by j0sh** [comment]
4s segment option is set in https://github.com/livepeer/go-livepeer/pull/351 now, thanks for catching that @yondonfu.

**Comment by j0sh** [comment]
Fixed in goclient by https://github.com/livepeer/go-livepeer/pull/351
